spec/controllers/services_controller_spec.rb

```diff
@@ -15,7 +15,29 @@
       expect(response).to have_http_status(:success)
     end
   end
-
+  describe "GET #edit" do
+    it "returns http success" do
+      get :edit, params: { id: service.id }
+      expect(response).to have_http_status(:success)
+    end
+  end
+  describe "GET #update" do
+    it 'expects ChangeServiceEvent to be called changing service details'  do
+      expect(ChangeServiceEvent).to receive(:create).and_call_original
+      patch :update, params: {
+        id: service.id,
+        service: { name: SecureRandom.alphanumeric, entity_id: "http://#{SecureRandom.uuid}.com" }
+      }
+      expect(subject).to redirect_to admin_path(anchor: :services)
```

Comment:
there should be an expectation whether it actually changed the thing

Reply:
I think this has been covered by visit_service_edit_page_spec

Reply:
ok added
```
 changed_service = ChangeServiceEvent.last.service
 expect(changed_service.entity_id).to eq entity_id
```